firmware/Board.h

```diff
@@ -6,12 +6,16 @@
 namespace board 
 {
   
+  struct Peripheral_t
+  {
+    int ENABLE;
+    int CURRENT;
+  };
+
```

Comment:
Can you explain a bit about how `Peripheral` and `Motor` relate?  It looks like some of the `Motor` functionality is abstracted into here.